### PR TITLE
Implement legacy config read/write

### PR DESCRIPTION
**Pull request by duncanspumpkin**
Should probably add a hook so that #913 is fixed for the config file.

### CODE REVIEW

File: src/OpenLoco/Config.cpp [hunk 1]
```diff
@@ -30,17 +32,70 @@ namespace OpenLoco::Config
...
+        {
+            _config->sound_quality = 0;
+            _config->vehicles_min_scale = 1;
+        }
+        _config->max_vehicle_sounds = _defaultMaxVehicleSounds[_config->sound_quality];
+        _config->max_sound_instances = _defaultMaxSoundInstances[_config->sound_quality];
+        for (size_t i = 0; i < std::size(_config->preferred_currency); ++i)
+        {
+            _config->preferred_currency[i] = _defaultPreferredCurrency[i];
+        }
```
Comment (AaronVanGeffen) [inline src/OpenLoco/Config.cpp:76]: memcpy?

File: src/OpenLoco/Config.cpp [hunk 2]
```diff
@@ -30,17 +32,70 @@ namespace OpenLoco::Config
...
+            uint32_t magicNumber{};
+            stream.read(reinterpret_cast<char*>(&magicNumber), sizeof(magicNumber));
+            if (magicNumber == 0x62272)
+            {
+                stream.read(reinterpret_cast<char*>(&*_config), sizeof(LocoConfig));
+                return _config;
+            }
+        }
+
+        if (_113E21C > 0x4000000)
```
Comment (AaronVanGeffen) [inline src/OpenLoco/Config.cpp:56]: Perhaps add a note here that from this point onward the defaults are being set.

File: src/OpenLoco/Config.cpp [hunk 3]
```diff
@@ -30,17 +32,70 @@ namespace OpenLoco::Config
         return _new_config;
     }
 
+    constexpr uint8_t _defaultMaxVehicleSounds[3] = { 4, 8, 16 };
+    constexpr uint8_t _defaultMaxSoundInstances[3] = { 6, 8, 10 };
+    constexpr uint8_t _defaultPreferredCurrency[16] = { 0x82, 0, 0, 0, 0x43, 0x55, 0x52, 0x52, 0x44, 0x4F, 0x4C, 0x4C, 0, 0, 0, 0 };
```
Comment (AaronVanGeffen) [inline src/OpenLoco/Config.cpp:37]: This reads CURRDOLL. Would it be possible to use e.g.:
```suggestion
    constexpr uint8_t _defaultPreferredCurrency[16] = { 0x82, 0, 0, 0, 'C', 'U', 'R', 'R', 'D', 'O', 'L', 'L', 0, 0, 0, 0 };
```

Comment (duncanspumpkin) [reply src/OpenLoco/Config.cpp:37]: What is currdoll?

Comment (IntelOrca) [reply src/OpenLoco/Config.cpp:37]: CURRency DOLLar?

Comment (duncanspumpkin) [reply src/OpenLoco/Config.cpp:37]: Ahh that makes a lot of sense.

**Review by AaronVanGeffen — APPROVED** [review]

**Comment by AaronVanGeffen** [comment]
We forgot to add hooks for config read/write.

**Pull request merged**